.github/workflows/waspc-build.yaml

```diff
@@ -0,0 +1,155 @@
+name: Build binaries
+
+# We never trigger this workflow directly.
+# We can call it manually (workflow_dispatch) or from other workflows (workflow_call).
+on:
+  workflow_dispatch:
+    inputs:
+      ghc-version:
+        description: "GHC version to use"
+        default: "8.10.7"
+        required: false
+      node-version:
+        description: "Node.js version to use"
+        default: "22"
+        required: false
+  workflow_call:
+    inputs:
+      ghc-version:
+        description: "GHC version to use"
+        default: "8.10.7"
+        type: string
+        required: false
+      node-version:
+        description: "Node.js version to use"
+        default: "22"
+        type: string
+        required: false
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+
+      matrix:
+        # == Why such a big, heterogeneous list? ==
+        # This is a bit of mish-mash of different platforms and architectures,
+        # so we need to build some of these directly in the runners, and some in
+        # containers. Each environment is a different OS and needs different
+        # dependencies.
+        env:
+          - name: linux-x86_64
+            runner: ubuntu-latest
+            container: ubuntu:20.04 # Old Ubuntu version so we have a low glibc requirement
+            install-deps: |
+              export DEBIAN_FRONTEND=noninteractive
+              apt-get update -y
+              # GHCup dependencies (https://www.haskell.org/ghcup/install/#version-2004-2010)
+              apt-get install -y build-essential curl libffi-dev libffi7 libgmp-dev libgmp10 libncurses-dev libncurses5 libtinfo5
+              # Cabal dependencies
+              apt-get install -y zlib1g-dev
+
+          # TODO: Add a Linux ARM64 build once we update the GHC version (#1446)
+          # GHC 8.10.7 does not support ARM64 on Linux yet
+
+          - name: linux-x86_64-static
+            runner: ubuntu-latest
+            # actions/setup-node does not work in alpine to work around this, we
+            # use the alpine variant of the official node image, which already
+            # has it installed.
+            container: node:${{ inputs.node-version }}-alpine
```

Comment:
I see we are not fixing the alpine version. From experience, that can result in trouble, due to latest version doing something different. We can protect from that by pinning down alpine version here, but risk that tehre won't be such version in combo with the inputs.node-version, or just leave it as it is and hope it won't cause issues. I am ok with leaving it as it is and seeing what happens, just wanted to be explicit about it.

Reply:
Yep, I am aware of this too.